### PR TITLE
feat: editable matrices for stratified model configs

### PR DESCRIPTION
**Pull request by shawnyama**
# Description

The values in the stratified matrices are editable. Something we should see about fixing is that the order of the configurations change sometimes - I assume this is elastic search related.

https://github.com/DARPA-ASKEM/Terarium/assets/28237258/9e9bc457-4fa1-42ac-b910-0562d80f671c

### CODE REVIEW

**Review by blanchco — COMMENTED** [review]
Left some notes to think about - but looks good!

File: packages/client/hmi-client/src/components/models/tera-stratified-model-configuration.vue [hunk 1]
```diff
@@ -153,6 +180,16 @@ const tableHeaders = computed<{ name: string; colspan: number }[]>(() => [
...
+// Makes cell inputs focus once they appear
+const vFocus = {
+	mounted: (el) => el.focus()
+};
+
+function updateModelConfigName(configIndex: number) {
+	modelConfigurations.value[configIndex].name = modelConfigInputValue.value;
+	updateModelConfiguration(modelConfigurations.value[configIndex]);
+}
+
```
Comment (blanchco) [inline packages/client/hmi-client/src/components/models/tera-stratified-model-configuration.vue:192]: Could be worth putting this in a separate util/service? Might be out of scope for this PR but I know the `tera-model-configuration` uses a similar function.  I imagine both these components use similar function to edit names/values as well.  Just a note

File: packages/client/hmi-client/src/utils/pivot.ts [hunk 2]
```diff
@@ -79,10 +79,35 @@ const pivotAxes = (data: any[], rowDimensions: string[], colDimensions: string[]
 	return { colAxis, rowAxis };
 };
 
+// Creates a M x 1 matrix where
+// M =  cardinality(rowDimensions[0]) * cardinality(rowDimensions[1]) * ... * cardinality(rowDimensions[m])
+export const createStateMatrix = (data: any[]) => {
+	const rows: any[] = [];
+
```
Comment (blanchco) [inline packages/client/hmi-client/src/utils/pivot.ts:86]: Are we able to type this (no any)?

Comment (mwdchang) [reply packages/client/hmi-client/src/utils/pivot.ts:86]: I think something is a little off here, we should be able to just use `createMatrx` to create a 1D matrix. 

Also utilities should not know anything about "states" or "transitions" - this is supposed to be a generalized pivot-table function.

File: packages/client/hmi-client/src/components/models/tera-stratified-model-configuration.vue [hunk 3]
```diff
@@ -175,6 +212,15 @@ function openValueModal(id: string, configIndex: number) {
 	}
 }
 
+function resetCellEditing() {
+	const row = { name: false };
```
Comment (mwdchang) [inline packages/client/hmi-client/src/components/models/tera-stratified-model-configuration.vue:216]: This data structure is a little weird, "name" is typically thought of as a string no? - what are we trying to convey?

File: packages/client/hmi-client/src/components/models/tera-stratified-model-configuration.vue (hunk 1 above)
Comment (shawnyama) [reply packages/client/hmi-client/src/components/models/tera-stratified-model-configuration.vue:192]: Yeah later I want to look into the viability of merging the two config components in some way. This component is made from a copy of the other config component - had to quickly get this version working during the hackathon.

File: packages/client/hmi-client/src/utils/pivot.ts (hunk 2 above)
Comment (shawnyama) [reply packages/client/hmi-client/src/utils/pivot.ts:86]: `createMatrix` ends up giving a 2D matrix for states, I'll double check why that is.

Comment (mwdchang) [reply packages/client/hmi-client/src/utils/pivot.ts:86]: Or rather it should create a 2D-matrix with one row in it.

File: packages/client/hmi-client/src/components/models/tera-stratified-model-configuration.vue [hunk 4]
```diff
@@ -175,6 +212,15 @@ function openValueModal(id: string, configIndex: number) {
...
+
+	// Can't use fill here because the same row object would be referenced throughout the array
+	const cellEditStatesArr = new Array(modelConfigurations.value.length);
```
[the remark this answers is not in the document]
Comment (shawnyama) [reply packages/client/hmi-client/src/components/models/tera-stratified-model-configuration.vue:219]: will keep note of this when I merge the configs components later

**Review by mwdchang — APPROVED** [review]

**Pull request merged**